File: MovieServer/MovieServer.API/MovieServer.Domain/src/ValueObjects/MovieItem.ts
import { MovieId, MovieName, MoviePrice, Quantity, Seat } from "@Domain/ValueObjects";
import { Aggregation } from "@Shared/Domain";

export default class MovieItem extends Aggregation {
    // movie Id
    // check
    constructor(
        public Id: MovieId,
        public Name: MovieName,
        public Price: MoviePrice,
        public Quantity: Quantity,
        public Seat: Seat) {
        super();
    }

    // method
    IncreaseQuantity(quantity: Quantity): void {
        this.Quantity.Increase(quantity.Value);
    }

    DecreaseQuantity(quantity: Quantity): void {
        this.Quantity.Decrease(quantity.Value);
    }
    ChangeQuantity(quantity: Quantity): void {
        this.Quantity.Change(quantity.Value);
    }
    ChangeSeat(seat: Seat): void {
        this.Seat.Change(seat.Value);
    }
}

File: OrderServer/OrderServer.API/OrderServer.Domain/ValueObjects/MovieItem.cs
using OrderServer.Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

//movie/add-to-cart

namespace OrderServer.Domain.ValueObjects
{
    public record MovieItem(Guid Id, string Name, float Price, string Seat, int Quantity);
}
